src/compiler/moduleSpecifiers.ts

```diff
@@ -706,11 +717,19 @@ namespace ts.moduleSpecifiers {
             let moduleFileToTry = path;
             if (host.fileExists(packageJsonPath)) {
                 const packageJsonContent = JSON.parse(host.readFile!(packageJsonPath)!);
-                // TODO: Inject `require` or `import` condition based on the intended import mode
                 if (getEmitModuleResolutionKind(options) === ModuleResolutionKind.Node12 || getEmitModuleResolutionKind(options) === ModuleResolutionKind.NodeNext) {
-                    const fromExports = packageJsonContent.exports && typeof packageJsonContent.name === "string" ? tryGetModuleNameFromExports(options, path, packageRootPath, packageJsonContent.name, packageJsonContent.exports, ["node", "types"]) : undefined;
+                    // `conditions` *could* be made to go against `importingSourceFile.impliedNodeFormat` if something wanted to generate
+                    // an ImportEqualsDeclaration in an ESM-implied file or an ImportCall in a CJS-implied file. But since this function is
+                    // usually called to conjure an import out of thin air, we don't have an existing usage to call `getModeForUsageAtIndex`
+                    // with, so for now we just stick with the mode of the file.
+                    const conditions = ["node", importingSourceFile.impliedNodeFormat === ModuleKind.ESNext ? "import" : "require", "types"];

```

Comment:
conditional-last is easier to read (as long as the order isn't otherwise important)
```suggestion
                    const conditions = ["node", "types", importingSourceFile.impliedNodeFormat === ModuleKind.ESNext ? "import" : "require"];
```

Reply:
I don’t _think_ the order is important, but I intentionally preserved the order that @weswigham consistently used elsewhere.

Reply:
(For future reference: no, order isn't important - it's essentially a set)